### PR TITLE
run e2e tests in parallel

### PR DESCRIPTION
**Pull request by wwwtyro**
@chuckbutler @mbruzek Ready for review

### CODE REVIEW

File: cluster/juju/layers/kubernetes-e2e/actions/test [hunk 1]
```diff
@@ -16,7 +17,6 @@ fi
 # get the host from the config file
 SERVER=$(cat /home/ubuntu/.kube/config | grep server | sed 's/    server://')
 
-# Execute the E2E test
-e2e.test -kubeconfig /home/ubuntu/.kube/config -host $SERVER -ginkgo.focus $FOCUS -ginkgo.skip $SKIP 2>&1 | tee /home/ubuntu/${JUJU_ACTION_UUID}.log
+ginkgo -nodes=$PARALLELISM $(which e2e.test) -- -kubeconfig /home/ubuntu/.kube/config -host $SERVER -ginkgo.focus $FOCUS -ginkgo.skip "$SKIP|\[Serial\]" 2>&1 | tee /home/ubuntu/${JUJU_ACTION_UUID}.log
```
Comment (marcoceppi) [inline cluster/juju/layers/kubernetes-e2e/actions/test]: I don't like the idea of force skipping serial, there's not effective way to E2E test for only serial which we still would want to do. Could we instead add `\[Serial\]` to default skip?

**Comment by marcoceppi** [comment]
Furthermore, I think we should have actions for each kind, with a then generic test action. For example, I would expect to see an action for `conformance` and then one for `conformance-serial` and finally the current `test` parameter. This isn't an issue with this PR, but something to consider going forward.

eg:
```
conformance:
  description: Parallel conformance test
  params:
    skip:
      description: Test to skip in addition to [Serial]
      type: string
      default:
    parallels:
      description: Number of test to runin paralell
      type: integer
      default: 25
```

**Comment by marcoceppi** [comment]
@Cynerva 
> Not sure what you mean.

I mean, we're always skipping Serial in the 'generic' test action. This means if it's set as focus it'll never actually run. In this way it's too implicit, instead of it being an explicit user defined item

**Comment by marcoceppi** [comment]
LGTM pending a review from someone else :+1:

File: cluster/juju/layers/kubernetes-e2e/layer.yaml [hunk 2]
```diff
@@ -4,6 +4,9 @@ includes:
   - layer:tls-client
   - interface:http
 options:
+  basic:
+    packages:
+      - golang-ginkgo-dev
```
Comment (mbruzek) [inline cluster/juju/layers/kubernetes-e2e/layer.yaml]: Does this work with the network limitations in the vpil lab? I assume it does.

File: cluster/juju/layers/kubernetes-e2e/actions/test [hunk 3]
```diff
@@ -16,7 +17,6 @@ fi
 # get the host from the config file
 SERVER=$(cat /home/ubuntu/.kube/config | grep server | sed 's/    server://')
 
-# Execute the E2E test
-e2e.test -kubeconfig /home/ubuntu/.kube/config -host $SERVER -ginkgo.focus $FOCUS -ginkgo.skip $SKIP 2>&1 | tee /home/ubuntu/${JUJU_ACTION_UUID}.log
+ginkgo -nodes=$PARALLELISM $(which e2e.test) -- -kubeconfig /home/ubuntu/.kube/config -host $SERVER -ginkgo.focus $FOCUS -ginkgo.skip "$SKIP" 2>&1 | tee /home/ubuntu/${JUJU_ACTION_UUID}.log
```
Comment (mbruzek) [inline cluster/juju/layers/kubernetes-e2e/actions/test]: It sounds like you were not able to get parallel runs without this extra ginkgo library. Is this an output of any kubernetes builds perhaps that we could also include in a resource for this charm?

**Comment by chuckbutler** [comment]
@marcoceppi i think you're on to something asking for scoped actions per test focus, but this is also very initial work. Restructuring of the action can go in a sep. PR. These changes post modification look good to me. I'm currently verifying and will 👍 once the test run is complete.

Thanks for the contributions @Cynerva and @wwwtyro

File: cluster/juju/layers/kubernetes-e2e/layer.yaml (hunk 2 above)
Comment (chuckbutler) [inline cluster/juju/layers/kubernetes-e2e/layer.yaml]: Yeah, archive access is available behind the egress fw. which is a +1 to this strategy. My concern lies with drift, but until its a problem its unfounded.

**Pull request merged**